src/app/basket.service.ts

```diff
@@ -328,37 +348,35 @@ export class BasketService {
                         part.indexOf('=') + 1,
                         part.lastIndexOf(']')
                     );
-                    let childObj = {};
-
-                    if (target) {
-                        childObj[ADDITIONALPROPS] = {
-                            'enterprise-id': target,
-                        };
-                    }
-
-                    object[container].forEach((child) => {
-                        if (child[keyName] === keyValue) {
-                            // console.log(
-                            //     'Found existing child',
-                            //     keyName,
-                            //     '=',
-                            //     keyValue
-                            // );
-                            childObj = child;
-                        }
-                    });
-                    if (!childObj[keyName]) {
+                    // TODO - taking a chance here - if key is numeric, then use it as a number rather than a string
+                    const valAsNum = parseInt(keyValue);
+                    if (isNaN(valAsNum)) {
+                        keyNames.set(keyName, keyValue);
                         childObj[keyName] = keyValue;
-                        object[container].push(childObj);
+                    } else {
+                        keyNames.set(keyName, valAsNum);
+                        childObj[keyName] = valAsNum;
+                    }
+                });
+            let alreadyExists = false;
+            object[container].forEach((child) => {
```

Comment:
These nested loops are a little confusing to me, but it as always been the case.
I wonder if breaking up this method in smaller pieces will make it easier to understand what we are doing (not urgent and not to be addressed in this PR)